packages/design-system/src/atoms/anchor/index.tsx

```diff
@@ -1,25 +1,40 @@
+import { Link } from 'native-base'
+import { ReactNode, useCallback } from 'react'
+import type { GestureResponderEvent } from 'react-native'
 import { TypographyBase, TypographyBaseProps } from '..'
 import { useTheme } from '../..'
 
 export type AnchorProps = {
-  children: string
-  href: string
+  children: ReactNode
+  href?: string
   target?: string
+  onPress?: () => void
 }
 
 export const Anchor = ({
   children,
   href,
   fontSize,
   target = '_blank',
+  onPress,
 }: AnchorProps & Pick<TypographyBaseProps, 'fontSize'>) => {
   const theme = useTheme()
 
+  const handlePress = useCallback(
+    (event?: GestureResponderEvent) => {
+      if (onPress && !href) {
+        event?.preventDefault()
+        onPress?.()
```

Comment:
I think at this point is sure onPress exists because it already pass the if statement on top 
```suggestion
        onPress()
```